test/unit/scratchOrgSettingsGeneratorTest.ts

```diff
@@ -567,22 +556,9 @@ describe('scratchOrgSettingsGenerator', () => {
       const settings = new SettingsGenerator();
       await settings.extract(scratchDef);
       await settings.createDeploy();
-      expect(addToZipStub.callCount).to.equal(2);
-      expect(addToZipStub.firstCall.firstArg)
-        .to.be.a('string')
-        .and.length.to.be.greaterThan(0)
-        .and.to.include('<RecordType/>');
-      expect(addToZipStub.firstCall.args[1]).to.include(path.join('objects', 'MyObject', 'MyObject.object'));
-      expect(addToZipStub.secondCall.firstArg)
-        .to.be.a('string')
-        .and.length.to.be.greaterThan(0)
-        .and.to.include('<RecordType>')
-        .and.to.include('<fullName>My-record-type</fullName>')
-        .and.to.include('<label>My-record-type</label>')
-        .and.to.include('<active>true</active>');
-      expect(addToZipStub.secondCall.args[1]).to.include(
-        path.join('objects', 'MyObject', 'recordTypes', 'My-record-type.recordType')
-      );
+      expect(addToZipStub.callCount).to.equal(1);
+      expect(addToZipStub.firstCall.firstArg).to.be.a('string').and.length.to.be.greaterThan(0);
+      expect(addToZipStub.firstCall.args[1]).to.include(path.join('objects', 'MyObject.object'));
```

Comment:
it's writing a file, but not verifying that the file is good.

If you tried to do all these permutations in NUTs, that would be a lot of scratch orgs and you'd have to find a way to check the object settings/record types etc.

that's why the "pure function" approach is good for stuff like this where there's a lot of input/output permutations.